influxdb3_wal/src/object_store.rs

```diff
@@ -450,21 +466,24 @@ impl FlushBuffer {
     /// responses. If a snapshot should occur with this flush, a semaphore permit is also returned.
     async fn flush_buffer_into_contents_and_responses(
         &mut self,
+        force_snapshot: bool,
     ) -> (
         WalContents,
         Vec<oneshot::Sender<WriteResult>>,
         Option<(SnapshotInfo, OwnedSemaphorePermit)>,
     ) {
         // convert into wal contents and responses and capture if a snapshot should be taken
         let (mut wal_contents, responses) = self.flush_buffer_with_responses();
-        self.snapshot_tracker.add_wal_period(WalPeriod {
-            wal_file_number: wal_contents.wal_file_number,
-            min_time: Timestamp::new(wal_contents.min_timestamp_ns),
-            max_time: Timestamp::new(wal_contents.max_timestamp_ns),
-        });
-
-        let snapshot = match self.snapshot_tracker.snapshot() {
+        if !force_snapshot {
```

Comment:
Why do we only add the WAL period if we're not forcing the snapshot? I think a comment here would be useful